### PR TITLE
[PE-3726] build: use eslint

### PR DESCRIPTION
**Pull request by luqven**
# Description
- Use ESLint for instead of JSCS and JSHint.
- Fix Node compact issue with fs promises.

# Steps to test
1. Clone this branch
2. Use a node version you'd like to test this version of web-tools in, `nvm use 14.0.0`
3. Run `npm link`
4. Open a terminal session in project you'd like to test this version in
5. Make sure the node version matches the `web-tools/` node version you linked from, `node --version` -> `14.0.0`
6. `npm link web-tools`
7. Run `npx gulp`

# Additional Info

Before/as this gets merged, all projects that currently lint with web-tools should have a lint-PR so that running web-tools in the future doesn't yield hundreds of indentation warnings (the one major change between our previous config and this one).

### CODE REVIEW

File: runcoms/.eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,149 @@
+{
+  "parserOptions": {
+    "ecmaVersion": 2020
+},
+"env": {
+    "es6": true
+},
```
Comment (mattlaguardia) [inline runcoms/.eslintrc.json:7]: is this how your prettier setup sets this file? example:
```
{
  "foo": {
    "bar": true
},
...
}

File: tools/pipelines/pipeline.js.js [hunk 2]
```diff
@@ -1,8 +1,7 @@
 var _ = require('lodash'),
     path = require('path'),
-    combine = require('stream-combiner'),
-    jshintReporter = require('reporter-plus/jshint'),
-    jscsReporter = require('reporter-plus/jscs');
+    eslint=require('gulp-eslint'),
+    combine = require('stream-combiner');
```
Comment (mattlaguardia) [inline tools/pipelines/pipeline.js.js:4]: OMG YES would love to throw reporter plus away

**Review by mattlaguardia — COMMENTED** [review]
Nice stuff here - LMK when its out of draft to re-review (if there are any changes etc)

File: runcoms/.eslintrc.json (hunk 1 above)
Comment (luqven) [reply runcoms/.eslintrc.json:7]: lmfao, no this is 💯 my mistake. Nice catch.

**Pull request merged**